src/OpenTelemetry.AutoInstrumentation/Loading/AspNetMetricsInitializer.cs

```diff
@@ -0,0 +1,49 @@
+// <copyright file="AspNetMetricsInitializer.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+#if NETFRAMEWORK
+
+using System;
+using System.Threading;
+
+namespace OpenTelemetry.AutoInstrumentation.Loading;
+
+internal class AspNetMetricsInitializer
+{
+    private int _initialized;
+
+    public AspNetMetricsInitializer(LazyInstrumentationLoader lazyInstrumentationLoader)
+    {
+        lazyInstrumentationLoader.Add(new AspNetMvcInitializer(InitializeOnFirstCall));
+        lazyInstrumentationLoader.Add(new AspNetWebApiInitializer(InitializeOnFirstCall));
+    }
+
+    private void InitializeOnFirstCall(ILifespanManager lifespanManager)
+    {
+        if (Interlocked.Exchange(ref _initialized, value: 1) != default)
+        {
+            // InitializeOnFirstCall() was already called before
+            return;
+        }
+
+        var instrumentationType = Type.GetType("OpenTelemetry.Instrumentation.AspNet.AspNetMetrics, OpenTelemetry.Instrumentation.AspNet");
```

Comment:
`instrumentationType` can be null

Reply:
That's our shipped library, I guess it could be null only when installation corrupts? - but then we have loads of other places unprotected.

Reply:
Indirect dependencies are the scenario that this may fail. Anyway, it is the same case as the other comment.